crates/amaru/tests/evaluate_ledger_states.rs

```diff
@@ -0,0 +1,223 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use amaru_kernel::{
+    AuxiliaryData, Epoch, EraHistory, Hasher, KeepRaw, MintedTransactionBody, MintedTx,
+    MintedWitnessSet, Point, PoolId, TransactionPointer, cbor, network::NetworkName,
+    protocol_parameters::ProtocolParameters,
+};
+use amaru_ledger::{
+    self,
+    context::DefaultValidationContext,
+    rules::transaction,
+    state::{self, volatile_db::StoreUpdate},
+    store::{EpochTransitionProgress, ReadStore, Store, TransactionalContext},
+};
+use amaru_stores::rocksdb::RocksDB;
+use pallas_addresses::Network;
+use pallas_codec::utils::AnyCbor;
+use std::{collections::BTreeSet, env, fs, iter, ops::Deref, path::PathBuf};
+
+use test_case::test_case;
+
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+struct TestContext {
+    ledger_dir: PathBuf,
+    pparams_dir: PathBuf,
+}
+
+fn get_test_context() -> Result<TestContext, Box<dyn std::error::Error>> {
+    let working_dir = env::current_dir()?;
+    Ok(TestContext {
+        ledger_dir: working_dir.join(PathBuf::from("../../ledger.db")),
+        pparams_dir: working_dir.join(PathBuf::from("../../cardano-blueprint/src/ledger/conformance-test-vectors/eras/conway/impl/dump/pparams-by-hash/")),
+    })
+}
+
+static TEST_MUTEX: Lazy<Mutex<()>> = Lazy::new(Mutex::default);
+
+include!("generated_ledger_conformance_test_cases.incl");
+
+fn evaluate_vector(snapshot: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let _shared = TEST_MUTEX.lock()?;
+    let network = NetworkName::Testnet(1);
+    let era_history = network.into();
+    match get_test_context() {
+        Ok(tc) => {
+            let vector_file = fs::read(snapshot)?;
+            let record: NESVector = minicbor::decode(&vector_file)?;
+            let () = import_vector(record, &tc.ledger_dir, era_history, &tc.pparams_dir)?;
+            Ok(())
+        }
+        Err(e) => {
+            tracing::warn!(
+                "skipping vector evaluation because some env vars are missing: {}",
+                e
+            );
+            Ok(())
+        }
+    }
+}
+
+fn peek_epoch(nes: &AnyCbor) -> Result<u64, Box<dyn std::error::Error>> {
+    let mut d = cbor::decode::Decoder::new(nes.raw_bytes());
+    let _ = d.array()?;
+    Ok(d.u64()?)
+}
+
+#[derive(cbor::Decode)]
+#[allow(dead_code)]
+struct NESVector {
+    #[n(0)]
+    initial_nes: AnyCbor,
+    #[n(1)]
+    final_nes: AnyCbor,
+    #[n(2)]
+    transactions: Vec<AnyCbor>,
+    #[n(3)]
+    test_state: String,
```

Comment:
```suggestion
    initial_state: AnyCbor,
    #[n(1)]
    final_state: AnyCbor,
    #[n(2)]
    transactions: Vec<AnyCbor>,
    #[n(3)]
    title: String,
```

Please avoid "cryptic" acronyms. "New Epoch State" is Haskell's jargon.